src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
+   auto data_size = data.size() * sizeof(std::pair<void*, unsigned int>);
+   buffer = nn_allocmsg(data_size, 0);
```

Comment:
_(A side note:

The "copy" policy reminds me of how g3log assures that LOG messages going into the logger are handled correct. Across the library API boundary the log message **has** to be copied. 

 I.e. different runtimes can allocate / deallocate ```std::string``` differently and across library runtimes it can be dangerous to receive and "own" std::strings and other memory-heap structures. It's a much better on Linux than on Windows. An example of this is compiling with full debug -O0 vs -Ofast_)